components/cam/src/physics/crm/sam/ADV_MPDATA/advect_scalar.F90

```diff
@@ -41,26 +50,39 @@ subroutine advect_scalar (ncrms,f,fadv,flux)
     else
       call advect_scalar2D(ncrms, f, u, w, rho, rhow, flux)
     endif
-
+#if defined(_OPENACC)
     !$acc parallel loop collapse(2) async(asyncid)
+#elif defined(_OPENMP)
```

Comment:
@xyuan , I think this is potentially a problem. `_OPENMP` is defined for any OpenMP-enabled compiler, but we only want this to be run if we're using OpenMP **4.5**. So I would recommend changing the define you pass to be `-D_OPENMP45` and to test for `#if defined(_OPENMP45)` instead of testing for `_OPENMP`.